tests/Doctrine/Tests/ORM/Functional/Ticket/DDC6029Test.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class DDC6029Test extends OrmFunctionalTestCase
+{
+    /**
+     * {@inheritDoc}
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->_schemaTool->createSchema([
+            $this->_em->getClassMetadata(DDC6029User::class),
+            $this->_em->getClassMetadata(DDC6029Group::class),
+            $this->_em->getClassMetadata(DDC6029Group2::class),
+        ]);
+    }
+
+    /**
+     * Verifies that when wrong entity is persisted via relationship field, the error message does not correctly state
+     * the expected class name.
+     */
+    public function testIssue()
+    {
+        $user = new DDC6029User();
```

Comment:
Shouldn't we add here the expected exception and the expected exception message? Like:

```php
$this->expectException(ORMInvalidArgumentException::class);
$this->expectExceptionMessage('Expected value of type "Doctrine\Tests\ORM\Functional\Ticket\DDC6029Group" for association field "Doctrine\Tests\ORM\Functional\Ticket\DDC6029User#$groups", got "Doctrine\Tests\ORM\Functional\Ticket\DDC6029Group2" instead.');
```